boards/posix/native_posix/cpu_wait.c

```diff
@@ -0,0 +1,75 @@
+/*
```

Comment:
Pitty, that this module now exists in two places, with the exact same contents...
Would be nice if they could have been merged at some common place, and the referenced to the relevant boards. But that is not the Zephyr way, it looks like.

Reply:
>with the exact same contents...

Not exactly. Beware that these functions rely on different "HW" provided only by the board. In the case of native_posix it is the same timer as for the system tick; in the case of the nrf52_bsim it is a fully dedicated timer (which does not exist in the real HW) just for this purpose.
But I kept the code and comments otherwise as close as possible so it could be merged eventually: We could move this functionality to the soc_inf code, by demanding common functionality from all POSIX boards (like we already do with other APIs), where they would provide a "HW" timer for this purpose with that kind of API.
But doing so now would break off-tree POSIX boards. I guess we can add that API later carefully and then move the code.